### PR TITLE
KAFKA-2675; SASL/Kerberos follow up

### PR DESCRIPTION
**Pull request by ijuma**

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java [hunk 1]
```diff
@@ -282,11 +282,10 @@
...
                                 .define(SaslConfigs.SASL_KERBEROS_TICKET_RENEW_WINDOW_FACTOR, Type.DOUBLE, SaslConfigs.DEFAULT_KERBEROS_TICKET_RENEW_WINDOW_FACTOR, Importance.LOW, SaslConfigs.SASL_KERBEROS_TICKET_RENEW_WINDOW_FACTOR_DOC)
                                 .define(SaslConfigs.SASL_KERBEROS_TICKET_RENEW_JITTER, Type.DOUBLE, SaslConfigs.DEFAULT_KERBEROS_TICKET_RENEW_JITTER, Importance.LOW, SaslConfigs.SASL_KERBEROS_TICKET_RENEW_JITTER_DOC)
                                 .define(SaslConfigs.SASL_KERBEROS_MIN_TIME_BEFORE_RELOGIN, Type.LONG, SaslConfigs.DEFAULT_KERBEROS_MIN_TIME_BEFORE_RELOGIN, Importance.LOW, SaslConfigs.SASL_KERBEROS_MIN_TIME_BEFORE_RELOGIN_DOC)
-                                .define(SaslConfigs.AUTH_TO_LOCAL, Type.LIST, SaslConfigs.DEFAULT_AUTH_TO_LOCAL, Importance.MEDIUM, SaslConfigs.AUTH_TO_LOCAL_DOC)
+                                .define(SaslConfigs.SASL_KERBEROS_PRINCIPAL_TO_LOCAL_RULES, Type.LIST, SaslConfigs.DEFAULT_SASL_KERBEROS_PRINCIPAL_TO_LOCAL_RULES, Importance.MEDIUM, SaslConfigs.SASL_KERBEROS_PRINCIPAL_TO_LOCAL_RULES_DOC)
```
Comment (harshach) [inline clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java]: We don't need this on client side. Auth to Local is useful on the broker side where we enforce authorization to convert a principal into a local. On client side we are not using this.

Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java]: I have removed the configs from the client side and use the `DEFAULT` rule in `KerberosNameParser` if the received parameter is null.

**Comment by junrao** [comment]
Thanks for the patch. Just one comment. In Login, we made the kafka-kerberos-refresh-thread a daemon thread. This means that we don't need to explicitly wait for it to shut down. However, it seems that LoginManager.release() still calls Login.shutdown() which waits for kafka-kerberos-refresh-thread to finish?

**Comment by junrao** [comment]
Also, you mentioned this covers https://github.com/apache/kafka/pull/359. However, in JaasUtils, we still reset Configuration to null in isZkSecurityEnabled()?

**Comment by ijuma** [comment]
My change in #359 was just for the SASL part, so I moved it here. I left the ZK part alone as I didn't want to affect Flavio's work.

**Comment by junrao** [comment]
Ok, it seems we can leave Login.shutdown() as it is. With kafka-kerberos-refresh-thread as a daemon thred, If the consumer client forgets to call close, then we can still shut down jvm.

Thanks for the patch. LGTM.

**Pull request closed without merging**